### PR TITLE
FPGA: Add `annotated_ptr` code sample

### PR DESCRIPTION
**Pull request by wangdi4**
# Add `annotated_ptr` code sample

## Description

Add a new FPGA code sample `annotated_ptr`  (from https://github.com/wangdi4/oneAPI-samples/pull/14)

## Checklist
Administrative
- [x] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): <insert Name Here>
- [ ] If you have any new dependencies/binaries, inform the oneAPI Code Samples Project Manager

Code Development
- [ ] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [x] Adhere to readme template 
- [x] Enforce format via clang-format config file
- [x] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [x] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [x] FPGA regtest:
   Linux: https://spetc.intel.com/testanalysis?trview=0&testRunIds=9513924&status=2&tapgsize=1000
Windows: https://spetc.intel.com/testanalysis?trview=0&testRunIds=9523592&tapgsize=1000

Security and Legal
- [ ] OSPDT Approval (see Project Manager for assistance)
- [x] Compile using the following compiler flags and fix any warnings, the falgs are: "/Wall -Wformat-security -Werror=format-security"
- [ ] Bandit Scans (Python only)
- [ ] Virus scan

Review
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth(@tomlenth) and/or Project Manager
- [ ] Tested using Dev Cloud when applicable

### CODE REVIEW

File: DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/annotated_ptr/src/annotated_ptr.cpp [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+#pragma unroll COLS
+      for (int j = 0; j < COLS; j++) sum += mat[j] * in_vec[j];
+
+      out_vec[i] = sum;
+    }
+  }
+};
+
+// verify results
+bool check_result(float *result, float *expected, int size) {
```
Comment (whitepau) [inline DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/annotated_ptr/src/annotated_ptr.cpp:80]:
```suggestion
bool CheckResult(float *result, float *expected, int size) {
```

Function names need to be `PascalCase`

**Comment by wangdi4** [comment]
@pmpeter1 @tomlenth could you help review this PR?

**Review by pmpeter1 — APPROVED** [review]

**Comment by tomlenth** [comment]
No comments on the README.  Very thorough.  I also took a look at the json to double check path. it all looks fine.

**Comment by wangdi4** [comment]
> No comments on the README. Very thorough. I also took a look at the json to double check path. it all looks fine.
> 
> actually, when looking at the other pull request, I noticed that the table for compilation type is not rendering correctly. both Windows and Linux need to be updated.

Thanks. The issue has been fixed.

**Review by whitepau — APPROVED** [review]

**Pull request merged**